### ISSUE TITLE
[Linux] open/save-file-menu doesn't remember last location on restart

### ISSUE THREAD

**Issue by FlailAway** [Ultimaker/Cura#5280]
**Application Version**
all so far!

**Platform**
Ubuntu/Mint

**Printer**
Not printer specific

**Steps to Reproduce**
Click "Save file " or "Open file"

**Actual Results**
Disassociated display of weird file names.
Nonsensical Most Recently Used file list.

**Expected results**
Opening at the last "Open" .obj etc Folder.
Saving at the last "Save" .obj etc Folder
Opening at the last "Open" .stl etc Folder.
Saving at the last "Save" .stl etc Folder

I suspect this all works better in windows but in Ubuntu/Mint it is abysmal. Every time I need to open an .obj file, I have to click, user-name, folder, folder, folder, file.obj

When I plug in a USB drive in the "Save to file" prompts changes to "Save to USB drive." Most, if not all printers these days allow navigation through folders for the .stl files. "Save to USB drvie," puts it in the root which is not helpful for clutter or re-use.

The entire file handling routines need to be gone through and made user friendly. But I guess for Linux users, who cares! {grin}

**Comment by rburema** [comment]
Hi @FlailAway,

You can click the arrow next to the 'Save to USB drive' to select the 'Save to File' option again, which should allow you to browse to the removable drive as well.

I've got a VM with Mint on it, and the 'recently used files' shows up just fine. Could you elaborate what you mean with nonsensical?

I can confirm that, when I restart, opening a file puts me in my home-folder again. I'll discuss that.

P.S. While it's true that Win-users of Cura form the majority, the dev. team itself purposefully is on a mix of Win, Mac & Linux (and not just on VM's either, I just had to use that since I'm on Win).

cheers, Remco.

**Comment by FlailAway** [comment]
Hi, thanks for the reply. It is not just restart, it is every time I open or save a file.

> You can click the arrow next to the 'Save to USB drive' to select the 'Save to File' option again, which should allow you to browse to the removable drive as well.

Yes, that's true, but when I take the USB out, use it in the printer and then plug it back in for the next item, it goes back to the "Save to USB drive and I have to drop it down and select ":Save to file."

> Could you elaborate what you mean with nonsensical?

I tried to drag and drop a pic("cura-02.jpg" 32K in size) but get a red error message below this text saying, "Something went really wrong and we can't process that file. Try again." Looks like the msoft  programmers are making things better here too. LOL! At the risk of an Einsteinium madness vote, I tried again and got the same message so I will ASCII it for you.

When I click the "Look in:" drop down, I get 

/tmp/.mount_KVkxjs/usr
/tmp/.mount_KVkxjs
/tmp
/
Recent Places
PLA
/home/user-name/Desktop
PLA
PLA

The "PLA" is a sub folder on the USB drive. Pretty sure that is a nonsensical display. :)

The only way I can get the .obj source folder to be visible in that drop-down is to save something to the folder I a getting th .obj from. Then when I click "open file" it will display it in the list for a few openings and then it is gone again.

There is too much to describe it all, it really needs someone to use it in real time and see just how bad it is. Not to mention my other gripe about the stack of "Reloads" in [https://github.com/Ultimaker/Cura/issues/5272](https://github.com/Ultimaker/Cura/issues/5272)

**Comment by rburema** [comment]
It's after work so I'll look at the rest some other time, however:
The 'too much reloads' (for mac/linux) will very probably be picked up during this sprint. Since it's treated as a bug, it might even be fixed in 4.0-BETA-2 (since bugs get around the feature-freeze).

**Comment by Ghostkeeper** [comment]
> I can confirm that, when I restart, opening a file puts me in my home-folder again. I'll discuss that.

This was a problem in the past, but I can't reproduce this any more on Master right now (running Ubuntu 18.10).

**Comment by Ghostkeeper** [comment]
> When I plug in a USB drive in the "Save to file" prompts changes to "Save to USB drive." Most, if not all printers these days allow navigation through folders for the .stl files. "Save to USB drvie," puts it in the root which is not helpful for clutter or re-use.

This is expected behaviour, for faster saving. If you want to save to a specific folder in the USB drive, you can simply use Save to File and select the correct folder in your USB drive.

**Comment by rburema** [comment]
@Ghostkeeper Yeah, I remember testing it with 3.6 yesterday, not master or 'current 4.0 from source'.

**Comment by Ghostkeeper** [comment]
> I tried to drag and drop a pic("cura-02.jpg" 32K in size) but get a red error message below this text saying, "Something went really wrong and we can't process that file. Try again." Looks like the msoft programmers are making things better here too. LOL! At the risk of an Einsteinium madness vote, I tried again and got the same message so I will ASCII it for you.

We don't have the words "try again" in our 3.6 source code anywhere. If you find an error message that is uninstructive, please indicate the correct text so that we can find it and correct it.

**Comment by Ghostkeeper** [comment]
> When I click the "Look in:" drop down, I get
> 
> /tmp/.mount_KVkxjs/usr
> /tmp/.mount_KVkxjs
> /tmp
> /
> Recent Places
> PLA
> /home/user-name/Desktop
> PLA
> PLA
> 
> The "PLA" is a sub folder on the USB drive. Pretty sure that is a nonsensical display. :)

Cura has no "look in" drop-down of any kind. Is that something your operating system adds?

**Comment by FlailAway** [comment]
@Ghostkeeper
> Cura has no "look in" drop-down of any kind. Is that something your operating system adds?

Dude, you need to slow down. It is the System File Open dialog that Cura displays. Well, it does on my versions of Ubuntu 16.04 and Mint 19.1 when using the Cura Appimage to Open or Save files.

>We don't have the words "try again" in our 3.6 source code anywhere.

Slooooow: :) Please look at the bottom of the window here on Github where you type in a reply. It has "Attach files by dragging ..." It was there I was trying to drag a .jpg file to show the nonsensical display. That's where I got the "try again." and did, but still could not attach the .jpg. Is it even possible to drag a .jpg into Cura? That's why I reverted to typing in the display I get.

**Comment by FlailAway** [comment]
> since bugs get around the feature-freeze

{grin} So abysmal-user friendliness for file handling is regarded as a bug? LOL

Thanks for your efforts, but it seems to this old fart that none of the Contributors actually use Cura for printing under Ubuntu. If they did, this would be fixed pronto. I have looked at Cura on win 8.1 and the file handling is a joy to use. But, it will be a cold day; before I go back to windows.

**Comment by nallath** [comment]
Ghostkeeper is actually using it with Ubuntu. I'm on Fedora myself.

**Comment by Ghostkeeper** [comment]
> Slooooow: :) Please look at the bottom of the window here on Github where you type in a reply. It has "Attach files by dragging ..." It was there I was trying to drag a .jpg file to show the nonsensical display. That's where I got the "try again." and did, but still could not attach the .jpg. Is it even possible to drag a .jpg into Cura? That's why I reverted to typing in the display I get.

Ah haha, I was thinking you were reporting another bug in Cura there but you were talking about the Github website. Sorry!

> Dude, you need to slow down. It is the System File Open dialog that Cura displays. Well, it does on my versions of Ubuntu 16.04 and Mint 19.1 when using the Cura Appimage to Open or Save files.

I figured out what you're seeing. Normally I see a dialogue like this:

![screenshot from 2019-02-12 10-58-03](https://user-images.githubusercontent.com/2448634/52627485-c6285100-2eb5-11e9-83c3-3282a990ad9f.png)

But that's only when running from source. When I run from the AppImage I see this:

![screenshot from 2019-02-12 11-05-37](https://user-images.githubusercontent.com/2448634/52627647-2a4b1500-2eb6-11e9-9a6f-8163ba71fe72.png)

That's probably more like what you're seeing. And it has this "Look in" field. It looks like this gets filled with the directories leading up to where the application is running. Due to the way that AppImage works, this is not the directory where the .AppImage file is located but the location where this gets mounted in `/tmp`. Probably Fedora users would also see this when using the AppImage.

I honestly wouldn't know what to do about it though.
